src/tree/v2/ResourceTreeDataProviderBase.ts

```diff
@@ -89,9 +93,59 @@ export abstract class ResourceTreeDataProviderBase extends vscode.Disposable imp
         return children;
     }
 
-    getParent(element: ResourceGroupsItem): vscode.ProviderResult<ResourceGroupsItem> {
+    getParent(element: ResourceGroupsItem): ResourceGroupsItem | undefined {
         return this.itemCache.getParentForItem(element);
     }
 
+    public isAncestorOf(element: ResourceGroupsItem, id: string): boolean {
+        return element?.isAncestorOf?.(id) || id.startsWith(this.getId(element) + '/');
+    }
+
+    public async findItem<T>(id: string): Promise<T | undefined> {
+        let element: ResourceGroupsItem | undefined = undefined;
+
+        outerLoop: while (true) {
+
+            const cachedChildren = this.itemCache.getChildrenForItem(element);
+            const children: ResourceGroupsItem[] | null | undefined = cachedChildren?.length ? cachedChildren : await this.getChildren(element);
+
+            if (!children) {
+                return;
+            }
+
+            for (const child of children) {
+                if (this.getId(child) === id) {
+                    return isWrapper(child) ? child.unwrap<T>() : child as unknown as T;
+                } else if (this.isAncestorOf(child, id)) {
+                    element = child;
+                    continue outerLoop;
+                }
+            }
+
+            return undefined;
+        }
+    }
+
     protected abstract onGetChildren(element?: ResourceGroupsItem | undefined): Promise<ResourceGroupsItem[] | null | undefined>;
+
+    private getId(element: ResourceGroupsItem): string {
+        let parent: ResourceGroupsItem | undefined = this.getParent(element);
+        let fullId = element.id;
+        if (!fullId.startsWith('/')) {
+            fullId = `/${fullId}`;
+        }
+        while (parent) {
+            let id = parent.id;
+            if (!id.startsWith('/')) {
+                id = `/${id}`;
+            }
+            // don't include grouping item IDs to the full ID
+            if (!id.startsWith('/groupings')) {
+                fullId = id + fullId;
+            }
```

Comment:
I special cased this for now. We could add a standard way for items to declare that their id should not contribute to the fullId of child items, but for now this works.